### PR TITLE
feat: [CDS-77798]: adding entity yaml in the response

### PR DESCRIPTION
**Pull request by lasdox**
## Describe your changes
Parsing the entityYaml response so we dont need to transverse the directory tree.

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`

### CODE REVIEW

**Comment by yogesh-chauhan** [comment]
@lasdox could you please update the PR description ? Thanks!

File: harness/cd/cac.go [hunk 1]
```diff
@@ -323,10 +323,24 @@ func (c *ConfigAsCodeClient) UpsertObject(input interface{}, filePath cac.YamlPa
...
-	if err != nil {
-		return err
+		if appId != "" && utils.HasField(responseObj, "ApplicationId") {
+			utils.MustSetField(responseObj, "ApplicationId", appId)
+		}
+		if err != nil {
+			return err
+		}
+
+	} else {
```
Comment (yogesh-chauhan) [inline harness/cd/cac.go:338]: do you think if there is any error/panic then we should just call ```c.FindObjectByPath(appId.(string), filePath, responseObj)```

Comment (lasdox) [reply harness/cd/cac.go:382]: Added panic recover

File: harness/cd/cac.go [hunk 2]
```diff
@@ -332,6 +332,64 @@ func (c *ConfigAsCodeClient) UpsertObject(input interface{}, filePath cac.YamlPa
...
 
+func (c *ConfigAsCodeClient) UpsertObject(input interface{}, filePath cac.YamlPath, responseObj interface{}) (err error) {
+	defer func(err error, input interface{}, filePath cac.YamlPath, responseObj interface{}) error {
+		if r := recover(); r != nil {
+			return c.UpsertObjectOld(input, filePath, responseObj)
+		}
+		return err
+	}(err, input, filePath, responseObj)
+
+	panic("test")
```
Comment (yogesh-chauhan) [inline harness/cd/cac.go:343]: is this intentional ?

**Review by micahlmartin — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Pull request merged**